google/auth/identity_pool.py

```diff
@@ -255,76 +265,93 @@ def __init__(
                 )
             self._credential_source_file = credential_source.get("file")
             self._credential_source_url = credential_source.get("url")
```

Comment:
Is it possible to use a data class for each credential types parameters? There is a lot of horizontal parameters / invariants now.

Reply:
Are you OK if that is done in a separate PR?

Reply:
Sounds good 👍